internal/tool/tool.go

```diff
@@ -137,6 +158,20 @@ func commandParameters(configFile *os.File, filesToAnalyse []string) []string {
 	return cmdParams
 }
 
+// TODO change this func
+func getFileLanguageFromExtension(filesToAnalyse []string) []string {
+	for _, file := range filesToAnalyse {
+		extension := filepath.Ext(file)
```

Comment:
I also can't think of another way of doing it
it's unfortunate that codacy-worker doesn't expose the language it's currently running the tool for...

a consequence of this is that the [File Extensions](https://docs.codacy.com/repositories-configure/file-extensions/) feature won't work. I don't think it's a major blocker, but something worth documenting as a limitation cc @nicklem

Reply:
@mrfyda Meaning that only Semgrep won't recognize custom extensions? I'm [mentioning it](https://tarot-2355-doc-new-tool-documentati--docs-codacy.netlify.app/repositories-configure/file-extensions/#configuring-file-extensions) on the relevant page, reviewing this with @claudiacarpinteiro.

Reply:
@nicklem As we talk, I would add this limitation to the [tool README](https://github.com/codacy/codacy-semgrep/blob/master/README.md), it seems the best place to fit it, although with low visibility.

For more visibility:

- I would also mention it on the [Configuring file extensions](https://tarot-2355-doc-new-tool-documentati--docs-codacy.netlify.app/repositories-configure/file-extensions/#configuring-file-extensions) page as you did, but not with high relevance, maybe as a bottom note. 
- Consider adding it also to [this table note](https://github.com/codacy/docs/pull/1921/files#diff-a23aa2ea454fb5eb6067478314f71cd05e09dd1cb53e8276d15171c60a08c606R534).